java-manta-it/src/test/java/com/joyent/manta/client/MantaClientSnapLinksIT.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (c) 2019, Joyent, Inc. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.joyent.manta.client;
+
+import com.joyent.manta.config.ConfigContext;
+import com.joyent.manta.config.IntegrationTestConfigContext;
+import com.joyent.manta.http.MantaHttpHeaders;
+import org.testng.Assert;
+import org.testng.annotations.*;
+
+import java.io.IOException;
+import java.util.UUID;
+
+import static com.joyent.manta.client.MantaClient.SEPARATOR;
+
+/**
+ * Tests the basic functionality of operations related to snaplinks
+ * for the {@link MantaClient} class.
+ *
+ * @author <a href="https://github.com/nairashwin952013">Ashwin A Nair</a>
+ */
+@Test(groups = "snaplinks")
+public class MantaClientSnapLinksIT {
+
+    private static final String TEST_DATA = "Arise,Awake And Do Not Stop Until Your Goal Is Reached.";
+
+    private MantaClient mantaClient;
+
+    private String testPathPrefix;
+
+    @BeforeClass
+    @Parameters({"usingEncryption"})
+    public void beforeClass(@Optional Boolean usingEncryption) throws IOException {
```

Comment:
Perhaps attempt to create a snaplink in the before method. If it fails with the message that snaplinks aren't supported, then we should throw a `SkipException` to indicate that the whole test class can't be run. This approach may have a problem, if so - I think the alternative is to throw the `SkipException` in each individual test.

Reply:
Noted.

Reply:
I am opting for the latter since I agree with you when you posited: *It will be relatively safer from the design perspective*. I have changed occurrences of ```putSnapLink()``` in the there tests that it was implemented in and didn't make any changes to the others, would you like me to include them in all ?

Reply:
A final change will also be added to the class description which will state the conditions under which this ```integration-test``` will be skipped.

Reply:
New changes passed in ```staging```.